.github/workflows/component-go.yaml

```diff
@@ -118,3 +123,32 @@ jobs:
         working-directory: "./examples/component/${{ matrix.example }}"
         run: |
           wash build
+
+      - name: run tests
+        working-directory: "./examples/component/${{ matrix.example }}"
+        run: go test .
+
+      # Run the wadm file and make sure it deploys
+      - name: test component load
+        shell: bash
+        working-directory: "./examples/component/${{ matrix.example }}"
+        # TODO: Add a test to the matrix for testing the running component (i.e. with `curl` or `wash call`)
+        run: |
+          set -xe
+          wash up &
+          WASH_PID=$!
+          sleep 4;
+          wash app deploy wadm.yaml;
+          TRIES=0
+          while [[ $(wash get inventory --output=json | jq '.inventories[0].components | length') -eq 0 ]] ; do
+            if [[ $TRIES -gt 10 ]]; then
+              echo "❌ failed to find component in inventory output after deploying example manifest";
+              exit -1;
+            fi
+            TRIES=$((TRIES+1));
+            sleep 1;
+          done;
+          echo "✅ successfully started at least one component";
+          wash app delete wadm.yaml;
+          kill $WASH_PID;
+          exit 0;
```

Comment:
Given how flaky this is in the main repo, I'm against bringing this logic over.

Reply:
I didn't think this was the flaky part of the tests, I thought that was mostly some of the wash tests

Reply:
[It definitely is flaky](https://github.com/wasmCloud/wasmCloud/actions/workflows/examples-components.yml?query=is%3Afailure)

Reply:
Ok, so here is what I did: We have a `--wadm-manifest` for `wash up` that ensures it is started for us. Out side of that, this bash script is pretty much what I'd do otherwise (in code I'd listen on a NATS sub, but this basically does the same thing). @joonas do you have other suggestions for how we could test this? I think testing that the component at least starts is an important smoke test, and like I mentioned in my future TODO we'd probably want to spin it up and test it

Reply:
Well...somehow that worked worse? Any suggestions on what you think we should do instead @joonas?

Reply:
Actually, it was because we were backgrounding rather than just letting wash do its thing. Now that I use the `-d` flag, wash makes sure the manifest is deployed and then we do the wait. Probably good enough for now, but I can increase the wait time for things to start as well if needed

Reply:
We can give it a go and see if it is better